Code/Framework/AzCore/AzCore/Math/UuidSerializer.cpp

```diff
@@ -129,4 +130,14 @@ namespace AZ
         
         return MessageResult("Default Uuid used.", JSR::ResultCode(JSR::Tasks::WriteValue, JSR::Outcomes::DefaultsUsed));
     }
+
+    bool JsonUuidSerializer::IsExplicitDefault(const rapidjson::Value& value)
+    {
+        if (value == m_zeroUuidString.c_str())
+        {
+            return true;
+        }
+
+        return BaseJsonSerializer::IsExplicitDefault(value);
+    }
```

Comment:
This function is not virtual by design so by adding it here it's going to result in unexpected behavior depending how the serializer is called. Since serializer are mostly called through the base pointer, this function shouldn't even be called. I would expect the content of this function as a check in the load function.

Reply:
That's weird, but ok I can move that check out.